ceno_zkvm/examples/fibonacci_elf.rs

```diff
@@ -0,0 +1,353 @@
+        .rev()
+        .find(|record| {
+            record.insn().codes().kind == EANY
+                && record.rs1().unwrap().value == CENO_PLATFORM.ecall_halt()
+        })
+        .and_then(|halt_record| halt_record.rs2())
+        .map(|rs2| rs2.value);
+
+    let final_access = vm.tracer().final_accesses();
+    let end_cycle: u32 = vm.tracer().cycle().try_into().unwrap();
+
+    let pi = PublicValues::new(
+        exit_code.unwrap_or(0),
+        vm.program().entry,
+        Tracer::SUBCYCLES_PER_INSN as u32,
+        vm.get_pc().into(),
+        end_cycle,
+        io_init.iter().map(|rec| rec.value).collect_vec(),
+    );
+
+    let mut zkvm_witness = ZKVMWitnesses::default();
+    // assign opcode circuits
+    let dummy_records = config
+        .assign_opcode_circuit(&zkvm_cs, &mut zkvm_witness, all_records)
+        .unwrap();
+    dummy_config
+        .assign_opcode_circuit(&zkvm_cs, &mut zkvm_witness, dummy_records)
+        .unwrap();
+    zkvm_witness.finalize_lk_multiplicities();
+
+    // Find the final register values and cycles.
+    let reg_final = reg_init
+        .iter()
+        .map(|rec| {
+            let index = rec.addr as usize;
+            if index < VMState::REG_COUNT {
+                let vma: WordAddr = CENO_PLATFORM.register_vma(index).into();
+                MemFinalRecord {
+                    addr: rec.addr,
+                    value: vm.peek_register(index),
+                    cycle: *final_access.get(&vma).unwrap_or(&0),
+                }
+            } else {
+                // The table is padded beyond the number of registers.
+                MemFinalRecord {
+                    addr: rec.addr,
+                    value: 0,
+                    cycle: 0,
+                }
+            }
+        })
+        .collect_vec();
+
+    // Find the final memory values and cycles.
+    let mem_final = mem_init
+        .iter()
+        .map(|rec| {
+            let vma: WordAddr = rec.addr.into();
+            MemFinalRecord {
+                addr: rec.addr,
+                value: vm.peek_memory(vma),
+                cycle: *final_access.get(&vma).unwrap_or(&0),
+            }
+        })
+        .collect_vec();
+    debug_memory_ranges(&vm, &mem_final);
+
+    // Find the final public IO cycles.
+    let io_final = io_init
+        .iter()
+        .map(|rec| *final_access.get(&rec.addr.into()).unwrap_or(&0))
+        .collect_vec();
+
+    // assign table circuits
+    config
+        .assign_table_circuit(&zkvm_cs, &mut zkvm_witness)
+        .unwrap();
+    mmu_config
+        .assign_table_circuit(
+            &zkvm_cs,
+            &mut zkvm_witness,
+            &reg_final,
+            &mem_final,
+            &io_final,
+        )
+        .unwrap();
+    // assign program circuit
+    zkvm_witness
+        .assign_table_circuit::<ExampleProgramTableCircuit<E>>(&zkvm_cs, &prog_config, vm.program())
+        .unwrap();
+
+    if std::env::var("MOCK_PROVING").is_ok() {
```

Comment:
`MOCK_PROVING=no ./run_my_programming` would also result in this branch being taken?  This might be a bit suprising.